### PR TITLE
Create plan section for azurerm_virtual_machine

### PR DESCRIPTION
**Pull request by msoni-tamr**

### CODE REVIEW

File: modules/tamr-vm/main.tf [hunk 1]
```diff
@@ -73,5 +73,16 @@ resource "azurerm_virtual_machine" "tamr-vm" {
     }
   }
 
+  plan {
+    count = contains(keys(var.plan), "name") &&
+            contains(keys(var.plan), "publisher") &&
+            contains(keys(var.plan), "product")
+            ? 1 : 0
+
+    name      = var.plan["name"]
```
Comment (souza-dan) [inline modules/tamr-vm/main.tf]: Is this ok when plan doesn't have a key name?

Do we need to do something like this?
```suggestion
    name      = contains(keys(var.plan), "name") ? var.plan["name"] : null
```

**Review by souza-dan — COMMENTED** [review]
Remember to increment version, update changelog, and run make terraform/docs.

Have you tested this out yet?

File: modules/tamr-vm/main.tf (hunk 1 above)
Comment (msoni-tamr) [reply modules/tamr-vm/main.tf]: The check on line 77 should prevent this block from getting created if name is not specified.

**Comment by msoni-tamr** [comment]
> Remember to increment version, update changelog, and run make terraform/docs.
> 
> Have you tested this out yet?

I haven't tested it out yet. Just wanted to create the PR to make sure things look reasonable. I'll test it out against the branch, then when things look good I'll increment the version and make the other changes.

File: modules/tamr-vm/main.tf (hunk 1 above)
Comment (souza-dan) [reply modules/tamr-vm/main.tf]: How did you confirm that setting the count to 0 doesn't evaluate these statements?

Comment (msoni-tamr) [reply modules/tamr-vm/main.tf]: Oh yeah, good point. I was incorrectly treating the block like a resource. Changed it to be a dynamic block.

Comment (msoni-tamr) [reply modules/tamr-vm/main.tf]: Turns out we're already have variables that capture all the info we need for the plan. Changing to use those.

**Pull request closed without merging**